File: tests/csharp/csharp_array_operations/array_find_returns_first_match.cs
// vybe-test: csharp/csharp_array_operations/array_find_returns_first_match
// origin: languages/csharp/tests/csharp/test_csharp_array_operations.rs

string __buf = "";

void __P(string s) {
    __buf = __buf + s + "\n";
}

void __Pr(string s) {
    __buf = __buf + s;
}

// The final WriteLine contributes a trailing newline that the expected line
// vector never carried, so BOTH forms are accepted.
void __Check(string want) {
    if (__buf != want && __buf != want + "\n") {
        Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
        throw new Exception("assertion failed");
    }
}

int[] a = {1,3,5,7};
__P((System.Array.Find(a, x => x > 3)).ToString());
__Check("5");

File: tests/js/array_copy_search_methods/array_find_returns_first_match.js
// vybe-test: js/array_copy_search_methods/array_find_returns_first_match
// origin: languages/js/tests/js/test_array_copy_search_methods.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

__check(__line([1,4,3].find(x=>x>2)), "4");
